bento/tools/_lp.py

```diff
@@ -332,6 +332,9 @@ def lp_diff_discrete(
         .reset_index()
     )
 
+    if diff_output.empty:
+        raise ValueError("No significant patterns found.")
```

Comment:
better to show a warning instead of raising an error. function should return immediately after warning. 
also I think we can deduce this before running diff_gene() by checking if all samples are the same pattern.